src/iostream.c

```diff
@@ -863,8 +863,10 @@ static Int InitKernel(
   /* init filters and functions                                          */
   InitHdlrFuncsFromTable( GVarFuncs );
   
+#if !defined(LIBGAP)
   /* Set up the trap to detect future dying children */
   signal( SIGCHLD, ChildStatusChanged );
+#endif
```

Comment:
As mentioned before (on the old PR): This `#ifdef` should be replaced by an `if (!disableSigChildHandler) { ...` or something like that, so that the same `libgap.so` can be used by GAP, and by Sage.